advanced_source/cpp_custom_ops.rst

```diff
@@ -0,0 +1,405 @@
+Custom C++ and CUDA Operators
+=============================
+
+.. note::
+   This tutorial is for PyTorch 2.4+ and the PyTorch nightlies.
+
+PyTorch offers a large library of operators that work on Tensors (e.g. torch.add, torch.sum, etc).
+However, you may wish to bring a new custom operator to PyTorch. This tutorial demonstrates the
+blessed path to authoring a custom operator written in C++/CUDA.
+
+For our tutorial, we’ll demonstrate how to author a fused multiply-add C++
+and CUDA operator that composes with PyTorch subsystems. The semantics of
+the operation are as follows:
+
+.. code-block:: python
+
+  def mymuladd(a: Tensor, b: Tensor, c: float):
+      return a * b + c
+
+You can find the end-to-end working example for this tutorial over at
+https://github.com/pytorch/extension-cpp .
+
```

Comment:
Can we add What you will learn  and prerequisites like [here](https://github.com/pytorch/tutorials/blob/main/beginner_source/template_tutorial.py#L9)?
Example:
What you will learn
* How to author and integrate custom operators written in C++ 
* How to compile custom C++/CUDA code and registering it within the PyTorch framework to make it accessible from Python.
* Advanced integration techniques

Prerequisites
* PyTorch 2.4 or later
* Read `torch.utils.cpp_extension <https://pytorch.org/docs/stable/cpp_extension.html>`_
* Basic understanding of C++ and CUDA programming